### PR TITLE
Development

### PR DESCRIPTION
**Pull request by nicolas-heigvd**
## docker-compose (v.1.27.4) integration using two distinct services:    

1. `db` to store a PostgreSQL 13.1 / PostGIS 3.1 database
2. `app` to store the application part, containing baremaps utils (based on `openjdk-11-jdk`) and based on a recent GDAL 3.2.1 image.

By doing this, baremaps images are not more useful, unless they embed hidden gems (?). 

\+ some minor modifications, fixes, and documentation update, the main ones concerning:
- [x] The `postgresql.conf` file was not taken into account.
- [x] The addition of a few SQL optimization queries (from ~2h > 10-15min for index creation on a supercomputer).
- [x] Add a `env.sample` file for convenience.
- [x] Make all connection strings use environment variables.

### ToDo:
- [x] The Natural Earth data set contains invalid self-intersecting geometries. It has been discarded because incompatible with ogr2ogr > 3.x See https://github.com/OSGeo/gdal/issues/2301
- [x] The Natural Earth data set import has been fixed.
- [x] Move scripts to a folder in the actual path, e.g. `/usr/local/bin`.
- [ ] ~Still missing some dynamic environment variable definition in `config.yaml` (Java doesn't seem to understand them and throws errors...)~
- [ ] ~Nice to have; a new flag, e.g. `--env` to `baremaps execute` so that we can pass environment variable to SQL scripts.~

### Info
DO NOT DELETE the `<development>` branch.    
DO NOT MERGE until WIP is resolved.

### CODE REVIEW

**Comment by bchapuis** [comment]
Looks really great! I started playing with the new configuration and I really like the idea of introducing different services. We may even split the gdal and baremaps imports in different services and use a minimal [image](https://github.com/baremaps/baremaps/blob/main/docker/baremaps/Dockerfile) for the latter.

Regarding the installation, I'm encountering an issue with the lan_access network, i.e., I'm unable to connect to the db from the host or from the app service. This may be due to the bridge driver, what do you think? What about using the host driver? 

Also, let me know if there are improvements to make to the baremaps CLI. For instance, I'm not sure it is usual to have environment variables interpolation in yaml or json. Do you have a particular use case in mind?

File: export.sh [hunk 1]
```diff
@@ -1,6 +1,9 @@
 #!/bin/bash
 
+HOST=${POSTGRES_HOST}
+PORT=${POSTGRES_PORT}
+
 baremaps export \
-  --database 'jdbc:postgresql://localhost:5432/osmvecto?allowMultiQueries=true&user=osmvecto&password=osmvecto' \
+  --database 'jdbc:postgresql://'${HOST}':'${PORT}'/'${POSTGRES_DB}'?allowMultiQueries=true&user='${POSTGRES_USER}'&password='${POSTGRES_PASSWORD} \
```
[the remark this answers is not in the document]
Comment (nicolas-heigvd) [reply export.sh]: Thanks. Fixed in `export.sh`, `import.sh` and `start.sh` by https://github.com/baremaps/openstreetmap-vecto/pull/19/commits/8e47b0ae5f2ee727168ade22519e32ccfb705d5e

**Comment by bchapuis** [comment]
I just had a second look at the [documentation](https://www.postgresql.org/docs/13/sql-createindex.html) and read the following about [CONCURRENTLY](https://www.postgresql.org/docs/13/sql-createindex.html#SQL-CREATEINDEX-CONCURRENTLY):

> When this option is used, PostgreSQL will build the index without taking any locks that prevent concurrent inserts, updates, or deletes on the table; whereas a standard index build locks out writes (but not reads) on the table until it's done.

In our case, we are fine with blocking the table for inserts, updates, or deletes as they will never occur. CONCURRENTLY may even result in a performance hit. Therefore, I would suggest that we remove all the CONCURRENLY from the queries. What do you think?

**Comment by bchapuis** [comment]
I also read the following article about index creatio, it suggests that the `max_parrallel_maintenance_workers` enable the creation of indexes in parallel. We should probably tweak this property and the following in our configuration file.

https://github.com/baremaps/openstreetmap-vecto/blob/10421ab0feda638c3d2f90b9d98e8df2fbb8e2cd/postgresql.conf#L172

I suspect that they have an effect similar to what is done in the `osm_prepare_indexes_comp.sql` file. What do you think?

https://github.com/baremaps/openstreetmap-vecto/blob/development/queries/osm_prepare_indexes_comp.sql

**Comment by nicolas-heigvd** [comment]
> I also read **_the following article_** about index creation, it suggests that the `max_parrallel_maintenance_workers` enable the creation of indexes in parallel. We should probably tweak this property and the following in our configuration file.
> 
> https://github.com/baremaps/openstreetmap-vecto/blob/10421ab0feda638c3d2f90b9d98e8df2fbb8e2cd/postgresql.conf#L172
> 
> I suspect that they have an effect similar to what is done in the `osm_prepare_indexes_comp.sql` file. What do you think?
> 
> https://github.com/baremaps/openstreetmap-vecto/blob/development/queries/osm_prepare_indexes_comp.sql

Which article exactly are you speaking about? I cannot see it?

**Comment by bchapuis** [comment]
Sorry, here is the article, but it looks like the two configuration parameters are combined, so my previous statement may be wrong.

https://www.cybertec-postgresql.com/en/postgresql-parallel-create-index-for-better-performance/

**Comment by nicolas-heigvd** [comment]
Ah, indeed, that's the article I was inspired from.

**Review by bchapuis — APPROVED** [review]

**Pull request merged**